src/common/libflux/Makefile.am

```diff
@@ -94,6 +94,8 @@ libflux_la_SOURCES = \
 	flog.c \
 	attr.c \
 	handle.c \
+	mpipe.c \
```

Comment:
commit message typo: triple-l in "internallly"

Reply:
Also, commit message nit, is "requeue" really part of a queue abstraction? Actually doesn't bother me too much but since it stood out to me thought I'd at least mention it (maybe it is a double-ended queue or "deque")? (But even if I'm not incorrect, it doesn't bother me enough to require any changes)

Reply:
Good point, I guess technically it's an [output-restricted deque](https://en.wikipedia.org/wiki/Double-ended_queue).

Maybe it should be renamed to `msg_deque` and (looking at the nice menu of methods in the wikipedia page) maybe `msg_deque_push_back`, `msg_deque_pop_front`, and `msg_deque_pop_front`?

Reply:
Eh, I thought enqueue, dequeue, and requeue, were fine and you could just update the description to a double ended queue if we wanted to be pedantic...

Reply:
Renaming stuff is easy - may as well get it right.  Are you OK if I change it to what was proposed above?  I think it's actually clearer that way now that you made me think about it a bit.

Reply:
I leave it up to you. I'm fine with the rename.